lib/active_merchant/billing/gateways/moneris.rb

```diff
@@ -331,7 +331,7 @@ def hashify_xml!(xml, response)
         return if xml.root.nil?
 
         xml.elements.each('//receipt/*') do |node|
-          response[node.name.underscore.to_sym] = normalize(node.text)
+          response[node.name.underscore.to_sym] = node.elements.count.zero? ? normalize(node.text) : hashify_children(node.elements)
```

Comment:
i think I saw some similar code in a different gateway but not sure which one to point you at :(